src/common/hooks/useConfig/useConfig.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+import { useConfig } from "./index";
+import { AsyncStorage } from "react-native";
+
+jest.mock("react-native", () => ({
+  AsyncStorage: {
+    setItem: jest.fn(),
+    getItem: jest.fn()
+  }
+}));
+
+const mockGetItem = AsyncStorage.getItem as jest.Mock;
+const mockSetItem = AsyncStorage.setItem as jest.Mock;
+
+describe("useConfig", () => {
+  beforeEach(() => {
+    mockGetItem.mockReset();
+    mockSetItem.mockReset();
+    mockGetItem.mockResolvedValue(JSON.stringify({ network: "mainnet" }));
+  });
+  it("should load saved config automatically", async () => {
+    expect.assertions(2);
+    const { result, waitForNextUpdate } = renderHook(() => useConfig());
+
+    await act(async () => {
+      await waitForNextUpdate();
+      await expect(result.current.loaded).toBe(true);
+      await expect(result.current.config).toStrictEqual({ network: "mainnet" });
+    });
+  });
+  it("should persist updated config", async () => {
+    expect.assertions(2);
+    const { result, waitForNextUpdate } = renderHook(() => useConfig());
+
+    await act(async () => {
+      await waitForNextUpdate();
+      await result.current.setValue("network", "ropsten");
+      await expect(result.current.config).toStrictEqual({ network: "ropsten" });
+      await expect(mockSetItem).toHaveBeenCalledWith(
+        "CONFIG",
+        JSON.stringify({ network: "ropsten" })
+      );
+    });
```

Comment:
Should we assert the previous value too?
Also, similar to the above, we don't need to nest everything:
```
    await waitForNextUpdate();

    act(() => {
      result.current.setValue("network", "ropsten");
    });

    expect(result.current.config).toStrictEqual({ network: "ropsten" });
    expect(mockSetItem).toHaveBeenCalledWith(
      "CONFIG",
      JSON.stringify({ network: "ropsten" })
    );
```

Reply:
I don' feel the need to assert previous value since this is testing for the logic of switching network, also it's tested in the previous test.